qiskit/providers/models/backendconfiguration.py

```diff
@@ -570,16 +584,77 @@ def acquire(self, qubit: int) -> AcquireChannel:
+                channel: int = None) -> List[ControlChannel]:
         """
         Return the secondary drive channel for the given qubit -- typically utilized for
         controlling multiqubit interactions. This channel is derived from other channels.
 
+        Args:
+            qubits: Tuple or list of qubits of the form `(control_qubit, target_qubit)`.
+            channel: Deprecated.
+
+        Raises:
+            BackendConfigurationError: If the ``qubits`` is not a part of the system.
+
         Returns:
-            Qubit control channel.
+            List of control channel.
         """
         # TODO: Determine this from the hamiltonian.
-        return ControlChannel(channel)
+        if channel is not None:
+            warnings.warn('control(channel: int) is deprecated. Use '
+                          'control(qubits: List or Tuple [control_qubit, target_qubit]) instead.',
+                          DeprecationWarning)
+            qubits = [channel]
+        try:
+            if isinstance(qubits, list):
+                qubits = tuple(qubits)
+            return self._control_channels[qubits]
+        except KeyError:
+            raise BackendConfigurationError("Couldn't find the ControlChannel operating on qubits "
+                                            "{} on {}-qubit system. The ControlChannel information"
+                                            " is retrieved from the "
+                                            " backend.".format(qubits, self.n_qubits))
+
+    def get_channel_qubits(self, channel: Channel) -> List[int]:
+        """
+        Return a list of indices for qubits which are operated on directly by the given ``channel``.
+
+        Raises:
+            BackendConfigurationError: If ``channel`` is not a found.
+
+        Returns:
+            List of qubits operated on my the given ``channel``.
+        """
+        try:
+            return self._channel_qubit_map[channel]
+        except KeyError:
+            raise BackendConfigurationError("Couldn't find the Channel - {}".format(channel))
+
+    def get_qubit_channels(self, qubit: Union[int, Iterable[int]]) -> List[Channel]:
+        r"""Return a list of channels which operate on the given ``qubit``.
+
+        Raises:
+            BackendConfigurationError: If ``qubit`` is not a found.
+
+        Returns:
+            List of ``Channel``\s operated on my the given ``qubit``.
+        """
+        channels = list()
+        try:
+            if isinstance(qubit, int):
+                for qbit in self._qubit_channel_map.keys():
+                    if qubit in qbit:
+                        channels.extend(self._qubit_channel_map[qbit])
+                qubit = [qubit]
+            if isinstance(qubit, list):
+                qubit = tuple(qubit)
+                channels.extend(self._qubit_channel_map[qubit])
+            elif isinstance(qubit, tuple):
+                channels.extend(self._qubit_channel_map[qubit])
```

Comment:
let's consolidate a little:
```suggestion
            if isinstance(qubit, int):
                for key in self._qubit_channel_map.keys():
                    if qubit in key:
                        channels.extend(self._qubit_channel_map[key])
                qubit = [qubit]
            if isinstance(qubit, list):
                qubit = tuple(qubit)
            if isinstance(qubit, tuple):
                channels.extend(self._qubit_channel_map[qubit])
```

Reply:
Not much of an improvement, take it if you like!

Reply:
Yes, `key` is way better than `qbit` 😄. Thanks! Fixed in [0da2877](https://github.com/Qiskit/qiskit-terra/pull/4097/commits/0da28775b838183eedd71d9157f20c08081e2833).